File: 217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 4/16/2026, 6:17:26 AM
1class Solution {
2    public boolean containsDuplicate(int[] nums) {
3        HashSet<Integer> seenNumbers = new HashSet<>();
4
5        for (int num : nums){
6            if (seenNumbers.contains(num)){
7                return true;
8            }
9            seenNumbers.add(num);
10        }
11        return false;
12    }
13}

File: 217-ContainsDuplicate/217-ContainsDuplicate.py
# Last updated: 16/08/2026, 13:04:27
1class Solution:
2    def containsDuplicate(self, nums: List[int]) -> bool:
3
4        seen = set()
5
6        for num in nums:
7            seen.add(num)
8
9        k = sorted(seen)
10
11        if len(k) == len(nums):
12            return False
13
14        return True
15        